defaults/main.yml

```diff
@@ -1560,8 +1718,70 @@ owncloud__apt_preferences__dependent_list_optional:
     pin: 'origin "download.owncloud.org"'
     priority: 995
     by_role: 'debops.owncloud'
-    state: '{{ owncloud__deploy_state }}'
+    state: '{{ "present"
+               if (owncloud__variant in ["owncloud"] and
+                   owncloud__deploy_state == "present")
+               else "absent" }}'
+
+
+# .. envvar:: owncloud__mariadb__dependent_databases [[[
+#
+# Configuration of the database managed by the debops.mariadb_ role.
+owncloud__mariadb__dependent_databases:
 
+  - database: '{{ owncloud__database_map[owncloud__database].dbname }}'
+    state: '{{ "present" if (owncloud__deploy_state != "purged") else "absent" }}'
+
+    # ## https://docs.nextcloud.com/server/12/admin_manual/installation/system_requirements.html#database-requirements-for-mysql-mariadb
+    # ## Currently experimental and not enabled by default.
+    # encoding: 'utf8mb4'
+    # collation: 'utf8mb4_general_ci'
+
+    # mariadb_server__options:
+    #   ## https://docs.nextcloud.com/server/11/admin_manual/installation/system_requirements.html#database-requirements-for-mysql-mariadb
+    #   'innodb_large_prefix': 'on'
+    #   'innodb_file_format': 'barracuda'
+    #   'innodb_file_per_table': 'true'
+
+                                                                   # ]]]
+# .. envvar:: owncloud__mariadb__dependent_users [[[
+#
+# Configuration of the database user managed by the debops.mariadb_ role.
+owncloud__mariadb__dependent_users:
+
+  - database: '{{ owncloud__database_map[owncloud__database].dbname }}'
+    user: '{{ owncloud__database_map[owncloud__database].dbuser }}'
+    password: '{{ owncloud__database_map[owncloud__database].dbpass }}'
+
+                                                                   # ]]]
+# .. envvar:: owncloud__postgresql__dependent_roles [[[
+#
+# Configuration of the database roles managed by the debops.postgresql_ role.
+owncloud__postgresql__dependent_roles:
+
+  - name: '{{ owncloud__database_name }}' # Separate role is needed when owncloud__database_name != owncloud__database_user
+  - name: '{{ owncloud__database_user }}' # Password is not passed directly - it will be read from the file
+
+                                                                   # ]]]
+# .. envvar:: owncloud__postgresql__dependent_groups [[[
+#
+# Configuration of the database groups managed by the debops.postgresql_ role.
+owncloud__postgresql__dependent_groups:
+
+  - roles: [ '{{ owncloud__database_user }}' ]
+    groups: [ '{{ owncloud__database_name }}' ]
+    database: '{{ owncloud__database_name }}'
+    enabled: '{{ owncloud__database_name != owncloud__database_user }}'
```

Comment:
The `enabled` parameter [is used in the debops.postgresql role](https://github.com/debops/ansible-postgresql/blob/master/tasks/main.yml#L155) but it doesn't seem to be properly documented. I guess this could be changed in `debops.postgres` role to `state` to keep it similar to other code like this. More upkeep to do...